packages/calcite-components/src/utils/component.ts

```diff
@@ -26,3 +26,28 @@ export function warnIfMissingRequiredProp<C extends LitElement>(
     logger.warn(`[${component.el.localName}] "${newProp.toString()}" or "${deprecatedProp.toString()}" is required.`);
   }
 }
+
+/**
+ * This utility logs a warning message when a property is missing and allows for a custom message to be provided.
+ *
+ * Note: this should only be used for cases where missing property or properties affects the component's functionality but is not required.
+ *
+ * @param component the component with the missing property
+ * @param prop the property that is missing
+ * @param message the custom message about functionality impact
+ * @param prop2 the second property that is missing
+ */
+export function warnIfMissingProps<C extends LitElement>(
+  component: C,
+  prop: keyof C,
+  message?: string,
```

Comment:
Let's remove the message argument. Messaging for required props should be consistent. If we need to provide different messaging, we could look at introducing a `context` argument to help with this. 

I still think we can tweak, and rename, `warnIfMissingRequiredProp` to achieve this.

Reply:
The props in question are not `required` properties. On the same note, created a different method to handle use case where missing props can affect functionality but not marked as `required`.

Reply:
I’m aware they’re not required, which is why I’m suggesting tweaking the existing utility (including renaming it). Both approaches aim to achieve the same goal: consistently warning developers about missing props that can’t be enforced via types.

I'm fine to proceed with the current approach, but let's revisit enhancing the util to handle this case.